src/GitTfs.VsCommon/TfsHelper.Common.cs

```diff
@@ -667,24 +661,35 @@ protected ITfsChangeset BuildTfsChangeset(Changeset changeset, IGitTfsRemote rem
             return tfsChangeset;
         }
 
-        protected virtual bool HasWorkItems(Changeset changeset)
+        protected virtual IEnumerable<TfsWorkitem> GetWorkItems(Changeset changeset)
         {
-            // This method wraps changeset.WorkItems, because
-            // changeset.WorkItems might result to ConnectionException: TF26175: Team Foundation Core Services attribute 'AttachmentServerUrl' not found.
-            // in this case assume that it is initialized to null
-            // NB: in VS2011 a new property appeared (AssociatedWorkItems), which works correctly
-            WorkItem[] result = null;
+            var workItems = changeset.AssociatedWorkItems.Select(wi => new TfsWorkitem
+            {
+                Id = wi.Id, 
+                Title = wi.Title
+            }).ToList();
+
             try
             {
-                result = Retry.Do(() => changeset.WorkItems);
+                var updatedWorkItems = changeset.WorkItems.Join(workItems, wi => wi.Id, twi => twi.Id, (wi, twi) =>
+                {
+                    twi.Description = wi.Description;
+                    twi.Url = Linking.GetArtifactUrl(wi.Uri.AbsoluteUri);
+                    return twi;
+                });
+
+                return updatedWorkItems;
             }
-            catch (ConnectionException exception)
+            catch (WorkItemTypeDeniedOrNotExistException e)
             {
-                if (!exception.Message.StartsWith("TF26175:"))
-                    throw;
+                Trace.TraceWarning(e.Message);
+                return workItems;
             }
+        }
 
-            return result != null && result.Length > 0;
+        protected virtual bool HasWorkItems(Changeset changeset)
```

Comment:
This method not longer needs to be overriden in any of its users, so either make it
```suggestion
        private bool HasWorkItems(Changeset changeset)
```
or better remove it altoegether and call GetWorkItems unconditonally and let it return an `Enumerable.Empty<ITfsWorkitem>()`  if there are none.